x-pack/solutions/security/plugins/security_solution/public/siem_migrations/rules/components/data_input_flyout/steps/hooks/use_missing_resources.ts

```diff
@@ -0,0 +1,59 @@
+/*
+ * Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
+ * or more contributor license agreements. Licensed under the Elastic License
+ * 2.0; you may not use this file except in compliance with the Elastic License
+ * 2.0.
+ */
+
+import { useCallback, useState } from 'react';
+import type { SiemMigrationResourceBase } from '../../../../../../../common/siem_migrations/model/common.gen';
+import type { UseMigrationStepsProps } from '../../../../../common/types';
+import { SplunkDataInputStep } from '../../../../../common/types';
+import type { RuleMigrationStats } from '../../../../types';
+import type { DashboardMigrationStats } from '../../../../../dashboards/types';
+
+interface MissingResourcesIndexed {
+  macros: string[];
+  lookups: string[];
+}
+
+export const useMissingResources = ({
+  setDataInputStep,
+}: {
+  setDataInputStep: UseMigrationStepsProps<
+    RuleMigrationStats | DashboardMigrationStats
+  >['setDataInputStep'];
+}) => {
+  const [missingResourcesIndexed, setMissingResourcesIndexed] = useState<
+    MissingResourcesIndexed | undefined
+  >();
+
+  const onMissingResourcesFetched = useCallback(
```

Comment:
Consider moving state and callback change in this FlyoutWrapper. This will make it possible for it to be available anywhere in the flyout via the context.

https://github.com/elastic/kibana/blob/ad533422325b457e7f4eac43120982e93377bbca/x-pack/solutions/security/plugins/security_solution/public/siem_migrations/rules/components/data_input_flyout/data_input_wrapper.tsx#L17

Reply:
Let's keep it as it is for now and we can revisit after feature freeze if needed.